virtio-devices/src/lib.rs

```diff
@@ -127,6 +131,37 @@ pub enum Error {
     ApplySeccompFilter(seccomp::Error),
 }
 
+#[derive(Clone, Copy, Debug, Default, Deserialize, Serialize, PartialEq)]
+pub struct TokenBucketConfig {
+    pub size: u64,
+    pub one_time_burst: Option<u64>,
+    pub refill_time: u64,
+}
+
+#[derive(Clone, Copy, Debug, Default, Deserialize, Serialize, PartialEq)]
+#[serde(deny_unknown_fields)]
+pub struct RateLimiterConfig {
+    pub bandwidth: Option<TokenBucketConfig>,
+    pub ops: Option<TokenBucketConfig>,
+}
+
+impl TryInto<rate_limiter::RateLimiter> for RateLimiterConfig {
```

Comment:
Is this why you had to define `RateLimiterConfig` and  `TokenBucketConfig` here and pull them in from config.rs rather than define them there and pull it into here?

Reply:
I thought that would introduce circular dependency issue, given both `virtio-devices` and `vmm` crates (where the `config.rs` comes from) need to access to those structs and the `vmm` crate depends on the `virtio-devices` crate. Please correct me if I am mistaken.

Reply:
Yeh, okay, I thought we already had config going into virtio-devices but I can see that we need it the way around you did it.

Reply:
I think the reason why we never ran into such thing so far is because we always process the config from DeviceManager, and then we create our virtio devices by passing values directly to them.
We never pass a configuration all the way down to the virtio device.